src/test/java/teammates/test/cases/datatransfer/InstructorAttributesTest.java

```diff
@@ -17,70 +17,116 @@
-
-        assertEquals(privileges, instructor1.privileges);
-
-        InstructorAttributes instructor2 =
-                new InstructorAttributes(instructor.googleId, instructor.courseId, instructor.name, instructor.email,
-                                         instructor.role, instructor.displayedName, instructor1.privileges);
+    @Test
+    public void testBuilderWithDefaultOptionalValues() {
+        InstructorAttributes instructor = InstructorAttributes
+                .builder("valid.google.id", "valid-course-id", "valid name", "valid@email.com")
+                .build();
+
+        // Check default values for optional params
+        assertEquals(DEFAULT_ROLE_NAME, instructor.role);
+        assertEquals(DEFAULT_DISPLAYED_NAME, instructor.displayedName);
+        assertEquals(DEFAULT_PRIVILEGES, instructor.privileges);
+        assertFalse(instructor.isArchived);
+        assertTrue(instructor.isDisplayedToStudents);
+    }
 
-        assertEquals(instructor1.privileges, instructor2.privileges);
+    @Test
+    public void testBuilderWithNullArguments() {
+        InstructorAttributes instructorWithNullValues = InstructorAttributes
+                .builder(null, null, null, null)
+                .withRole(null)
+                .withDisplayedName(null)
+                .withIsArchived(null)
+                .withPrivileges((InstructorPrivileges) null)
+                .build();
+        // No default values for required params
+        assertNull(instructorWithNullValues.googleId);
+        assertNull(instructorWithNullValues.courseId);
+        assertNull(instructorWithNullValues.name);
+        assertNull(instructorWithNullValues.email);
+
+        // Check default values for optional params
+        assertEquals(DEFAULT_ROLE_NAME, instructorWithNullValues.role);
+        assertEquals(DEFAULT_DISPLAYED_NAME, instructorWithNullValues.displayedName);
+        assertEquals(DEFAULT_PRIVILEGES, instructorWithNullValues.privileges);
+        assertFalse(instructorWithNullValues.isArchived);
+    }
 
-        InstructorAttributes instructorNew =
-                new InstructorAttributes(instructor.googleId, instructor.courseId, instructor.name, instructor.email,
-                                         instructor.role, false, instructor.displayedName, instructor1.privileges);
+    @Test
+    public void testBuilderCopy() {
+        InstructorAttributes instructor = InstructorAttributes
+                .builder("valid.google.id", "valid-course-id", "valid name", "valid@email.com")
+                .withDisplayedName("Original instructor")
+                .build();
+
+        InstructorAttributes instructorCopy = InstructorAttributes
+                .builder(instructor.googleId, instructor.courseId, instructor.name, instructor.email)
+                .withDisplayedName(instructor.displayedName)
+                .build();
+
+        assertEquals(instructor.googleId, instructorCopy.googleId);
```

Comment:
Pls hard code the expected value.
e.g `assertEquals(instructor.googleId, "valid.google.id");`